From f597431fcb05487865e0239710bf0e54e50c0d61 Mon Sep 17 00:00:00 2001
From: Stella Laurenzo <stellaraccident@gmail.com>
Date: Fri, 9 May 2025 19:21:34 -0700
Subject: [PATCH 1/3] Use is_versioned = true consistently in both
 Comgr::LoadLib paths.

Prior to this, RTCProgram was using is_version = true and device initialization was using false. On Windows, this was causing us to attempt to load amd_comgr_3.dll at device load and amd_comgr0605.dll for RTC programs. Obviously, there is only one DLL and it must be used consistently.
---
 rocclr/device/device.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rocclr/device/device.cpp b/rocclr/device/device.cpp
index 2154535c1..fd8511ca2 100644
--- a/rocclr/device/device.cpp
+++ b/rocclr/device/device.cpp
@@ -754,7 +754,7 @@ bool Device::ValidateComgr() {
 #if defined(USE_COMGR_LIBRARY)
   // Check if Lightning compiler was requested
   if (settings_->useLightning_) {
-    constexpr bool kComgrVersioned = false;
+    constexpr bool kComgrVersioned = true;
     std::call_once(amd::Comgr::initialized, amd::Comgr::LoadLib, kComgrVersioned);
     // Use Lightning only if it's available
     settings_->useLightning_ = amd::Comgr::IsReady();
-- 
2.49.0.windows.1

